### PR TITLE
Migrate Rust API from beamy repository

### PR DESCRIPTION
**Pull request by carlmartus**
Rust API created by @niclaslind for previous _Beamy labs repository_.

Migrating to this repository to ensure maintainance.

## Changes
Apart from moving the code. There are some adaptations to the code.

1. Use shared proto-files instead of local copies.
2. Refactor examples by integrating them into Crate examples. Instead of being their own crates.
3. Add _github action_ to make sure the code compiles on latest stable version of Rust.

### CODE REVIEW

File: rust/remotivelabs-broker/src/remotive_api/base.rs [hunk 1]
```diff
@@ -0,0 +1,1260 @@
+//// Symbolic message representing no argument needed or provided.
```
Comment (JohanKarlbergg) [inline rust/remotivelabs-broker/src/remotive_api/base.rs:1]: Should we maybe remove this file and add it to .gitignore?

File: rust/remotivelabs-broker/src/remotive_api/mod.rs
Comment (JohanKarlbergg) [inline rust/remotivelabs-broker/src/remotive_api/mod.rs:1]: Is this also output from proto generation?

File: rust/remotivelabs-broker/src/remotive_api/base.rs (hunk 1 above)
Comment (carlmartus) [reply rust/remotivelabs-broker/src/remotive_api/base.rs]: Removed file

File: rust/remotivelabs-broker/src/remotive_api/mod.rs
Comment (carlmartus) [reply rust/remotivelabs-broker/src/remotive_api/mod.rs:1]: No this one is necesary to export the bindings.

File: rust/remotivelabs-broker/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    {
+        if let Some(entry) = entry.path().to_str() {
+            upload_file(system_stub, entry, entry.replace(path, "")).await?;
+        }
+    }
+
+    Ok(())
+}
+
+/// Reload beamybroker configuration
```
Comment (jankronquist) [inline rust/remotivelabs-broker/src/lib.rs:103]: beamybroker -> RemotiveBroker all in places

File: rust/remotivelabs-broker/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    let sha256 = get_sha256(path)?;
+    let chunk_size = 1000000;
+
+    // Make the path to unix style
+    let unix_destination_path = Path::new(&dest_path).to_slash().unwrap();
+
+    let upload_iterator = generate_data(path, unix_destination_path, chunk_size, sha256).await?;
+    let _response = system_stub.upload_file(upload_iterator).await?;
+
+    println!("Uploaded file {}", path);
```
Comment (jankronquist) [inline rust/remotivelabs-broker/src/lib.rs:73]: if this is a library we should probably not println anything

**Comment by jankronquist** [comment]
Add description run to run examples, eg `cargo run --example pub_client`

File: rust/remotivelabs-broker/src/lib.rs (hunk 2 above)
Comment (carlmartus) [reply rust/remotivelabs-broker/src/lib.rs:103]: Replaced text.

File: rust/remotivelabs-broker/src/lib.rs (hunk 3 above)
Comment (carlmartus) [reply rust/remotivelabs-broker/src/lib.rs:73]: Refactoring

**Comment by carlmartus** [comment]
> Add description run to run examples, eg `cargo run --example pub_client`

Added subsection to `README.md`.

File: rust/remotivelabs-broker/README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# RemotiveLabs Broker
+
+Rust library with gRPC-bindings and utility functions.
+
+gRPC bindings are generated from [this source](../../proto/) when compiling the project.
+
+This project is contributed by _Niclas Lind_ (@niclaslind).
+
+## Publish / Subscribe example
+See `examples/pub_client.rs` and `examples/sub_client.rs` for a sample on how to use the library.
```
Comment (jankronquist) [inline rust/remotivelabs-broker/README.md:10]: mention that this example is using the virtual namespace

File: rust/remotivelabs-broker/examples/configuration/interfaces.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{
+  "default_namespace": "VirtualInterface",
+  "chains": [
+    {
+      "type": "virtual",
+      "device_name": "virtual",
+      "namespace": "VirtualInterface"
+    }
+  ],
+  "gateway": {
```
Comment (jankronquist) [inline rust/remotivelabs-broker/examples/configuration/interfaces.json:10]: can we simplify this configuration? ie remove things that are not used?

File: rust/remotivelabs-broker/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    let data = Some(Data::Chunk(buf));
+
+    // Create the upload requests with file description and data
+    let file_description = FileUploadRequest { data: fd };
+    let data = FileUploadRequest { data };
+
+    Ok(stream::iter(vec![file_description, data]))
+}
+
+/// Upload file to BeamyBroker
```
Comment (jankronquist) [inline rust/remotivelabs-broker/src/lib.rs:58]: BeamyBroker

Comment (carlmartus) [reply rust/remotivelabs-broker/src/lib.rs:58]: Fixed

File: rust/remotivelabs-broker/examples/configuration/interfaces.json (hunk 5 above)
Comment (carlmartus) [reply rust/remotivelabs-broker/examples/configuration/interfaces.json:10]: Removed unused parameters.

File: rust/remotivelabs-broker/README.md (hunk 4 above)
Comment (carlmartus) [reply rust/remotivelabs-broker/README.md:10]: Added mention

**Comment by carlmartus** [comment]
X-API key added.

**Review by jankronquist — APPROVED** [review]

**Pull request merged**